### ISSUE TITLE
Небольшой глюк

### ISSUE THREAD

**Issue by wd4** [alexandernst/angular-multi-select#35]
Здравствуйте. Использую ваш плагин. Но проблема вот в чем:
У меня есть, например, модель:

``` javascript
[{
name:'Test'
 sub: [
{name: 'Вася'},
{name: 'Петя'}
]
},
{
name:'Test2'
 sub: [
{name: 'Коля'},
{name: 'Вася'}

]
}
]

Когда я делаю поиск по имени Вася, то все Васи найдены.
Когда я стираю из строки поиска то, что ввел, то во второй группе Коля заменяется Васей.


Даже тут http://alexandernst.github.io/angular-multi-select/#/demo-grouping
Если ввести несколько раз слово Chrome, в первой группе все будет заменено на Chrome. При бэкспейсе - то же самое.
```

**Comment by alexandernst** [comment]
Thank you for the report. I'm already working on it.

Sorry for not replying in Russian, I don't have enough knowledge of that language in order to reply you properly :)

**Comment by wd4** [comment]
Sorry for russian. You made cool script. I think that bug is.. may be in part of code.. $scope.$watch('filterModel'...)

**Comment by alexandernst** [comment]
Fixed in https://github.com/alexandernst/angular-multi-select/commit/58690d8a395e375435e61e2b9ec971a58f8a095d :+1: 
I released `5.3.16` on npm.
